server/functions.php

```diff
@@ -303,3 +303,84 @@ function get_user_by_ssn($ssn) {
 		);
 	}
 }
+
+if(!function_exists("get_lectures_by_course")){
+	function get_lectures_by_course($courseId, $startTs = null, $endTs = null){
+		$pdo = new PDO("sqlite:../db.sqlite");
+
+		// Get type of user
+		$userData = get_myself();
+
+		if (!$userData) {
+			// User doesn't exist, but is logged in ❓❓❓
+			return;
+		}
+
+		$userType = intval($userData['type']);
+
+		if($userType != USER_TYPE_SPRT_OFCR){
+			throw new ErrorException("Wrong permissions");
+		}
+
+		$query = 'SELECT L.*, C.name AS courseName, C.code AS courseCode, R.name AS roomName FROM lectures AS L, rooms AS R, (SELECT name,code FROM courses WHERE ID = :courseId) AS C WHERE L.room_id = R.ID AND L.course_id = :courseId';
```

Comment:
Does this return also cancelled lectures? if yes, there  should be a field in the returned object